### PR TITLE
dns/dnscrypt-proxy: added support for 'disabled_server_names'

### PR DESCRIPTION
**Pull request by Karlson2k**
A simple patch which add ability to disable unwanted servers from automatic selection.
Servers may be unstable, return incorrect data, be suspected in logging, or user may just don't like some specific servers. This patch gives ability to easily exclude any server(s) while maintaining ability to dynamically update the list of servers.

### CODE REVIEW

**Comment by Karlson2k** [comment]
@mimugmail I'm not sure that `HostnameField` is correct. Names are not really host names, they are just identifiers, like "ev-to", "dnscrypt.ca-2", "scaleway-fr".
Does `HostnameField` work correctly with such names?

**Comment by mimugmail** [comment]
You are totally right, I didn't check the syntax (dont use this plugin in production).
Maybe you can add a validation to model like this:

`<mask>/^([0-9a-zA-Z\-\_]){1,64}$/u</mask>`

**Comment by Karlson2k** [comment]
@mimugmail Updated by adding commit with correction.
If update is fine, I can squash down to the single commit.

**Comment by Karlson2k** [comment]
@mimugmail Should I squash into a single commit?

**Comment by fichtner** [comment]
No need, GitHub can do it :)

File: dns/dnscrypt-proxy/src/opnsense/mvc/app/models/OPNsense/Dnscryptproxy/General.xml [hunk 1]
```diff
@@ -136,5 +136,12 @@
         <serverlist type="CSVListField">
             <Required>N</Required>
         </serverlist>
+        <disabled_serverlist type="TextField">
+            <mask>/^([a-z0-9.\-]{1,70}$/</mask>
```
Comment (Karlson2k) [inline dns/dnscrypt-proxy/src/opnsense/mvc/app/models/OPNsense/Dnscryptproxy/General.xml:140]: US ASCII chars are used only, so no Unicode flag here.
No uppercase letters. Currently names can be up to 48 characters long, so I used `70` here to have some space.

Comment (Karlson2k) [reply dns/dnscrypt-proxy/src/opnsense/mvc/app/models/OPNsense/Dnscryptproxy/General.xml:140]: Missing closing comma here. Need to update

**Comment by Karlson2k** [comment]
@mimugmail Updated

**Comment by Karlson2k** [comment]
@fichtner @mimugmail Are any changes needed?

**Comment by mimugmail** [comment]
You need to add `","` to allowed chars as it's a CSVList :)

**Comment by Karlson2k** [comment]
@mimugmail Updated

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**